buildbot-host/buildmaster/master.cfg

```diff
@@ -504,10 +505,15 @@ for factory_name, factory in factories.items():
 # We need to create schedulers after the builders, because otherwise the build
 # name lists are not available yet.
 c['schedulers'] = []
+
+# Only build if any of the files in the Change match this regular expression
+openvpn_filter_fn = r'^(src/(openvpn)/(\w|-)+\.(c|h)|m4/(\w|-)+\.m4|compat.m4|configure.ac|Makefile.am|version.m4)$'
+
 c['schedulers'].append(schedulers.SingleBranchScheduler(
                         name="openvpn-default",
                         change_filter=util.ChangeFilter(branch  = openvpn_branch,
-                                                        project = 'openvpn'),
+                                                        project = 'openvpn',
+                                                        filter_fn = lambda c: any([re.search(openvpn_filter_fn, f) for f in c.files])),
```

Comment:
Honestly this scheduler does a build every few days currently. So not really worth implementing it here. The Gerrit scheduler would probably be the more interesting use case here.

Reply:
Ok. I can't test the Gerrit scheduler easily, but I think the filter can be moved over there without any modifications. The Change object should look the same anyways.

Reply:
Just saw on https://docs.buildbot.net/latest/manual/configuration/changesources.html#gerriteventlogpoller that we probably need to add get_files argument to GerritEventLogPoller. Just noting it here to have it recorded it somewhere.